jpype/protocol.py

```diff
@@ -0,0 +1,92 @@
+import _jpype
+import datetime
+import sys
+import _jpype
+from . import _jclass
+from . import _jcustomizer
+
+# Copies of all private base types for reference
+_JClass = _jpype._JClass
+_JObject = _jpype._JObject
+_JException = _jpype._JException
+_JNumberLong = _jpype._JNumberLong
+_JNumberFloat = _jpype._JNumberFloat
+_JComparable = _jpype._JComparable
+_JChar = _jpype._JChar
+_JBoolean = _jpype._JBoolean
+_JArray = _jpype._JArray
+_JBuffer = _jpype._JBuffer
+
+
+if sys.version_info < (3, 8):
+    # Before 3.8 we have to use attribute type converters
+    from collections.abc import Sequence, Mapping
+
+    class _AttributeMeta(type):
+        def __instancecheck__(self, obj):
+            return hasattr(obj, self._attrib)
+    SupportsPath = _AttributeMeta("SupportsPath", (object,), {
+                                  "_attrib": "__fspath__"})
+    SupportsIndex = _AttributeMeta("SupportsIndex", (object,), {
+                                   "_attrib": "__index__"})
+    SupportsFloat = _AttributeMeta("SupportsFlaot", (object,), {
+                                   "_attrib": "__float__"})
+else:
+    from typing import Protocol, runtime_checkable, abstractmethod
+    from typing import SupportsIndex, SupportsFloat
+    from typing import Sequence, Mapping
+
+    # Types we need
+    @runtime_checkable
+    class SupportsPath(Protocol):
+        """An ABC with one abstract method __fspath__."""
+        __slots__ = ()
+
+        @abstractmethod
+        def __fspath__(self) -> str:
+            pass
+
+
+@_jcustomizer.JConversion("java.nio.file.Path", instanceof=SupportsPath)
+def _JPathConvert(jcls, obj):
+    Paths = _jpype.JClass("java.nio.file.Paths")
+    return Paths.get(obj.__fspath__())
+
+
+@_jcustomizer.JConversion("java.io.File", instanceof=SupportsPath)
+def _JFileConvert(jcls, obj):
+    return jcls(obj.__fspath__())
+
+
+@_jcustomizer.JConversion("java.util.Collection", instanceof=Sequence)
+def _JSequenceConvert(jcls, obj):
+    return _jclass.JClass('java.util.Arrays').asList(obj)
+
+
+@_jcustomizer.JConversion("java.util.Collection", none=str)
```

Comment:
This decorator looks out of place here ...

Reply:
Moved back  to resolve.